Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -8,55 +10,63 @@ public GildedRose(Item[] items) {
+        {
+            if(items[days].name.equals("Sulfuras, Hand of Ragnaros")) {
+                updateHandOfRagnaros(items[days]);
+            } else
+            if(items[days].name.equals("Aged Brie")) {
+                updateAgedBrie(items[days]);    
+            } else
+            if(items[days].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
+                updateBackstagepasses(items[days]);    
             } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
-
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-
-                        if (items[i].sellIn < 6) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-                    }
-                }
+                updateEverythingElse(items[days]);
             }
+        }
+    }
 
-            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                items[i].sellIn = items[i].sellIn - 1;
-            }
+    private void updateHandOfRagnaros(Item item){
+    
+    }
+
+    private void updateAgedBrie(Item item){
+        if (item.quality > 50)
+            return;
+        
+        item.sellIn = item.sellIn - 1;
+        item.quality = item.quality + 1;
+        if (item.sellIn < 0){
+            item.quality = item.quality + 1;    
+        }   
+    }
 
-            if (items[i].sellIn < 0) {
-                if (!items[i].name.equals("Aged Brie")) {
-                    if (!items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].quality > 0) {
-                            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                                items[i].quality = items[i].quality - 1;
-                            }
-                        }
-                    } else {
-                        items[i].quality = items[i].quality - items[i].quality;
-                    }
-                } else {
-                    if (items[i].quality < 50) {
-                        items[i].quality = items[i].quality + 1;
-                    }
-                }
+    private void updateBackstagepasses(Item item){
+        if (item.quality < 50) {
+            item.quality = item.quality + 1;
+            if (item.sellIn < 11 && item.quality < 50){
+                item.quality = item.quality + 1;
             }
+
+            if (item.sellIn < 6 && item.quality < 50) {
+                item.quality = item.quality + 1;  
+            }   
+        }
+        item.sellIn = item.sellIn - 1;
```

Comment:
das ruft ihr in jeder update methode auf (außer sulfuras), sollte man evtl. extrahieren